bundles/org.openhab.voice.watsonstt/src/main/java/org/openhab/voice/watsonstt/internal/WatsonSTTService.java

```diff
@@ -134,8 +140,13 @@ public STTServiceHandle recognize(STTListener sttListener, AudioStream audioStre
         final AtomicReference<@Nullable WebSocket> socketRef = new AtomicReference<>();
         final AtomicBoolean aborted = new AtomicBoolean(false);
         executor.submit(() -> {
-            socketRef.set(speechToText.recognizeUsingWebSocket(wsOptions,
-                    new TranscriptionListener(socketRef, sttListener, config, aborted)));
+            var speechToText = this.speechToText;
+            if (speechToText != null) {
```

Comment:
Yes, the compiler reports it as needed.

Reply:
This is probably because you are using this.speechToText.
Can you please try this:
```
       var stt = this.speechToText;
        if (stt == null) {
            throw new STTException("service is not correctly configured");
        }
...
       executor.submit(() -> {
            socketRef.set(stt.recognizeUsingWebSocket(wsOptions,
                    new TranscriptionListener(socketRef, sttListener, config, aborted)));
        });
```

Reply:
You are right, I though that was what I tried, thank you. 
Changed!